datalab/datalab_session/file_utils.py

```diff
@@ -112,3 +109,18 @@ def scale_points(height_1: int, width_1: int, height_2: int, width_2: int, x_poi
     x_points = width_2 - x_points
 
   return x_points, y_points
+
+def create_output(cache_key, np_array=None, fits_file=None, large_jpg=None, small_jpg=None, index=None, comment=None):
```

Comment:
I think we're doing a lot here for the programmer, and there's a ton of combinations of args here that can result in errors that aren't being handled. What happens, for instance, if np_array is none and fits_file is none? Presumably save_fits_and_thumbnails will be called and it will error out. There's nothing indicating to the developer that certain args in certain combinations are required, nor is it clear what exceptions will be raised.

I think we can create a class that does some input validation, throws errors if things are mal-defined and has some nice helper methods to abstract some of the work away from the developer - maybe we can whiteboard this out tomorrow?